src/commands.rs

```diff
@@ -89,6 +89,13 @@ fn run_server_process(startup_timeout: Option<Duration>) -> Result<ServerStartup
     let runtime = Runtime::new()?;
     let exe_path = env::current_exe()?;
     let workdir = exe_path.parent().expect("executable path has no parent?!");
+
+    // Spawn a blocking task to bind the Unix socket
+    let sp = socket_path.clone();
+    let listener = runtime.block_on(async move {
+        tokio::net::UnixListener::bind(sp).expect("Failed to bind Unix socket")
+    });
```

Comment:
```suggestion
    let listener = runtime.block_on(async move {
        tokio::net::UnixListener::bind(sp)
    })?;
```
Untested, but perhaps returning the error directly would be better (to preserve the original functionality).

Reply:
Works well, I've pushed a fixup with this change.